hudi-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadata.java

```diff
@@ -388,6 +504,21 @@ private void initIfNeeded() {
     return Pair.of(logRecordScanner, logScannerOpenMs);
   }
 
+  // NOTE: We're allowing eager full-scan of the log-files only for "files" partition.
+  //       Other partitions (like "column_stats", "bloom_filters") will have to be fetched
+  //       t/h point-lookups
+  private boolean isFullScanAllowedForPartition(String partitionName) {
```

Comment:
for FILES, can we honor scan log files config in HoodieMetadataConfig.

Reply:
it's done as you suggested and added a task to handle all partitions https://issues.apache.org/jira/browse/HUDI-3809